src/Build/Logging/BaseConsoleLogger.cs

```diff
@@ -332,9 +265,9 @@ internal void ShowPerfSummary()
         /// </summary>
         internal void WritePretty(int indentLevel, string formattedString)
         {
-            StringBuilder result = new StringBuilder();
```

Comment:
Since I was a little surprised this was done elsewhere in the file (but not here), I tested the scenario.

| Method | Time (ms) | Time (ticks) |
|----|-------|---|
| String formatting | 14   | 51400      |
| StringBuilder(calculated length) | 11  | 41400      |
| StringBuilder() | 14  | 49183      |

Seems like a significant win. This is for 100,000 iterations though. And this code is a really hot path when in diag logging. But to overall build time it's not going to be significant. So you're probably right not worth it, but it's an easy calculation and measurably faster :).